stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+        self.input_folder = input_folder
+        self.output_folder = output_folder
+
+        # all the .obj, .glb, .jpg (textures)
+        self.ext_resource = ""
+        self.ext_resource_id = 1
+
+        # SpatialMaterials (construction of textures)
+        self.sub_resource = ""
+        self.sub_resource_id = 1
+
+        # MeshInstances and collision scenes (.glb)
+        self.nodes = f'[node name="{output_folder}" type="Spatial"]\n'
+        if scale != 1:
+            self.nodes += f'transform = Transform( {scale}, 0, 0, 0, {scale}, 0, 0, 0, {scale}, 0, 0, 0 )\n'
+        self.nodes += "\n" # add extra new line to prepare for next node
+
+        # keep track of textures so we don't generate them more than once
+        self.texture_index = dict()
+
+    def add_texture(self, texture):
+        # don't generate textures more than once
+        if texture not in self.texture_index:
+            # handle PBR textures (require .jpgs)
+            if texture in Textures.texture_dict:
+                t_info = Textures.texture_dict[texture]
+                self.sub_resource += f'[sub_resource type="SpatialMaterial" id={self.sub_resource_id}]\n'
+
+                for jpg, value in t_info.jpg_dict.items():
+                    if jpg in ["albedo", "roughness", "metallic"]:
+                        self.sub_resource += f"{jpg}_texture = ExtResource( {self.ext_resource_id} )\n"
+                    elif jpg == "normal":
+                        self.sub_resource += Textures.NORMAL_OPTIONS + f"normal_texture = ExtResource( {self.ext_resource_id} )\n"
+                    elif jpg == "depth":
+                        self.sub_resource += Textures.DEPTH_OPTIONS + f"depth_texture = ExtResource( {self.ext_resource_id} )\n"
+                    # add .jpg as ext_resource
+                    self.ext_resource += f'[ext_resource path="res://{Textures.TEXTURE_FOLDER}/{t_info.folder}/{value}" type="Texture" id={self.ext_resource_id}]\n'
+                    self.ext_resource_id += 1
+
+                # add uv1_scale if texture has it
+                if t_info.uv1_scale:
+                    self.sub_resource += f"uv1_scale = {t_info.uv1_scale}\n"
+
+            # handle textures which don't need .jpgs (only need sub_resource)
+            elif texture in Textures.other_textures:
+                self.sub_resource += f'[sub_resource type="SpatialMaterial" id={self.sub_resource_id}]\n{Textures.other_textures[texture]}'
+
+            self.sub_resource += "\n"
+                
+            # add the sub resource's index to texture_index so we can use it later 
+            self.texture_index[texture] = self.sub_resource_id
+            self.sub_resource_id += 1 
+
+    def add_obj_file(self, stl_file_extless, uv_map, mesh_compression, texture):
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .stl into bpy
+        bpy.ops.import_mesh.stl(filepath = f"{self.input_folder}/{stl_file_extless}.stl")
```

Comment:
`pathlib.Path` would allow you to combine directory and file strings in a manner specific to any OS without having to introduce OS-specific syntax.
```suggestion
        import_path = Path(self.input_folder) / Path(stl_file_extless).with_suffix('.stl')
        bpy.ops.import_mesh.stl(filepath=str(import_path))
```